### DISCUSSION TITLE
SSR and loading indicators from the docs

### DISCUSSION THREAD

**Discussion by nehalist** [trpc/trpc#3082] [Q&A]
Guess I'm missing something here, but unfortunately, I'm having a hard time with the docs: in https://trpc.io/docs/v10/nextjs the example utils/trcp.ts sets `ssr: true`, but the pages/index.tsx example uses

```
  const hello = trpc.hello.useQuery({ text: 'client' });
  if (!hello.data) {
    return <div>Loading...</div>;
  }
```
 

Since SSR would prevent the page from being rendered until the query is done, how would this loading indicator ever be shown?

**Comment by KATT** [comment]
When you navigate to this route on the client

**Reply by nehalist** [reply]
So that I understand correctly: I longer need getServerSideProps at all (since I can't even use it anymore) and do everything via trpc `useQuery` in the component that gets rendered server-side on initial request and client-side on navigation?
